### PR TITLE
snapd: updates

### PR DESCRIPTION
**Pull request by bboozzoo**
@aimileus thanks for updating the package to 2.31!

@zyga @aimileus some fixes and updates:

-  add missing build dependency on libseccomp

- add bash-completion as optional dependency

- drop python-docutils from makedepends (we don't use them, not sure why they
  were left behind)

- update mkversion to include $pkgrel, this affects what is reported in `snap
  version`:
```
    snap    2.31-2
    snapd   2.31-2
    series  16
    arch
    kernel  4.15.1-2-ARCH
```
- dump extra cleanups in pre_remove(), those are already done by snap-mgmt

- pass all configuration settings when running `make -C data`

### CODE REVIEW

**Comment by bboozzoo** [comment]
Let me do a quick amend, there's some code in snapd.install that can go away now.

File: .SRCINFO [hunk 1]
```diff
@@ -1,21 +1,22 @@
...
 	url = https://github.com/snapcore/snapd
 	install = snapd.install
 	arch = x86_64
 	license = GPL3
 	makedepends = git
 	makedepends = go-pie
 	makedepends = go-tools
+	makedepends = libseccomp
 	makedepends = libcap
-	makedepends = python-docutils
```
Comment (zyga) [inline .SRCINFO:13]: This was used for creating manual pages for snap-confine.

File: .SRCINFO [hunk 2]
```diff
@@ -1,21 +1,22 @@
...
 	makedepends = systemd
 	makedepends = xfsprogs
 	depends = squashfs-tools
 	depends = libseccomp
 	depends = libsystemd
+	optdepends = bash-completion: bash completion support
```
Comment (zyga) [inline .SRCINFO:19]: Not a strong feeling but does it have to be optional? Is it such a big dependency to avoid?

**Review by zyga — APPROVED** [review]
LGTM with an open question on the python build dependency.

**Comment by bboozzoo** [comment]
hope you don't mind guys, i'll be pushing this to AUR

**Comment by bboozzoo** [comment]
Pushed as https://aur.archlinux.org/cgit/aur.git/commit/?h=snapd&id=57cf6441a75d1d01e0c720bdb77f875e87bd4dc7

**Pull request closed without merging**